code/core/src/shared/test-provider-store/index.ts

```diff
@@ -408,22 +412,39 @@ export function createTestProviderStore({
     setFullState: universalTestProviderStore.setState,
     onSettingsChanged: (listener) =>
       universalTestProviderStore.subscribe('settings-changed', listener),
-    runAll: () => universalTestProviderStore.send({ type: 'run-all' }),
-    clearAll: () => universalTestProviderStore.send({ type: 'clear-all' }),
+    runAll: () => {
+      if (universalTestProviderStore.status !== UniversalStore.Status.READY) {
+        universalTestProviderStore.untilReady().then(() => {
+          universalTestProviderStore.send({ type: 'run-all' });
+        });
+      } else {
+        universalTestProviderStore.send({ type: 'run-all' });
+      }
```

Comment:
Doesn't untilReady just resolve immediately if it's already ready? Seems convoluted to do an extra if/else condition.

Reply:
Even when a promise is already resolved, the content of `then` will still be pushed onto the microtask queue. But I agree, that's probably fine here, and not worth the extra if statement.